### PR TITLE
fix(redoc): Upgrade redoc CLI

### PR DESCRIPTION
**Pull request by federicojasson**
# Description

Replace deprecated `redoc-cli` by `@redocly/cli`.
This change doesn't affect the outcome of the build commands, except of `build:redoc` (checked).
Reviewing commit by commit should help understand the changes.

Concrete changes:

- Move options to config file `.redocly.yaml`
- Extra: fix script `start:public`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: .redocly.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  hideDownloadButton: true
+  hideSingleRequestSampleTab: true
+  jsonSampleExpandLevel: 4
+  pathInMiddlePanel: true
+  requiredPropsFirst: true
+  sortPropsAlphabetically: true
+  theme:
+    colors:
+      primary:
+        main: teal
```
Comment (federicojasson) [inline .redocly.yaml:13]: Reference: https://redocly.com/docs/api-reference-docs/configuration/functionality/
Note: `noAutoAuth` is deprecated, so it wasn't included here.

File: package.json [hunk 2]
```diff
@@ -12,11 +12,11 @@
...
+    "start": "redocly preview-docs reference/openapi.yaml",
+    "start:public": "redocly preview-docs public-tmp/openapi.yaml",
     "build": "yarn build:openapi; yarn build:redoc",
     "build:openapi": "node ./scripts/build-openapi.js reference build",
-    "build:redoc": "rm -rf redoc; mkdir redoc; redoc-cli bundle --options. --output redoc/index.html reference/openapi.yaml --options.theme.colors.primary.main=teal --options.expandResponses=all --options.hideDownloadButton=true --options.hideSingleRequestSampleTab=true  --options.expandSingleSchemaField=true --options.jsonSampleExpandLevel=4 --options.noAutoAuth=true --options.pathInMiddlePanel=true --options.requiredPropsFirst=true --options.sortPropsAlphabetically=true",
+    "build:redoc": "rm -rf redoc; mkdir redoc; redocly bundle --output redoc/redoc.yaml reference/openapi.yaml",
```
Comment (federicojasson) [inline package.json:19]: The output is no longer an HTML file, but a YAML file.

File: .redocly.yaml (hunk 1 above)
Comment (gryevns) [reply .redocly.yaml:13]: Is `noAutoAuth` replaced by another config option or does the build include no auth by default now?

Comment (federicojasson) [reply .redocly.yaml:13]: From the [changelog](https://redocly.com/docs/api-reference-docs/changelog/#deprecated-showsecurityschemetype), there's no other option to replace it.

If I'm not mistaken, that option hides the `Authorizations` section, which we're already showing in https://gr4vy.com/docs/reference

Screen Shot 2022-10-25 at 16 52 04

Comment (gryevns) [reply .redocly.yaml:13]: There is a `hideSecuritySection` option documented at https://redocly.com/docs/api-reference-docs/configuration/functionality/ - although not sure if this is the same.

Comment (federicojasson) [reply .redocly.yaml:13]: Nice finding. That hides the `Authorizations` section. I'm not sure if we want that or not though 🤔 
@cbetta

Comment (gryevns) [reply .redocly.yaml:13]: Currently the `BearerAuth` link doesn't actually link to anything - probs best to hide it completely or add an auth section.

**Review by gryevns — APPROVED** [review]

**Pull request merged**